pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountThetaSketchAggregationFunction.java

```diff
@@ -19,136 +19,144 @@
+      Preconditions.checkArgument(paramsExpression.getType() == ExpressionContext.Type.LITERAL,
+          "Second argument of DISTINCT_COUNT_THETA_SKETCH aggregation function must be literal (parameters)");
+      Parameters parameters = new Parameters(paramsExpression.getLiteral());
+      int nominalEntries = parameters.getNominalEntries();
+      _updateSketchBuilder.setNominalEntries(nominalEntries);
+      _setOperationBuilder.setNominalEntries(nominalEntries);
+    }
 
-    // NOTE: This function expects at least 4 arguments: theta-sketch column, nominalEntries, predicate(s), post-aggregation expression.
-    Preconditions.checkArgument(numArguments > 3,
-        "DistinctCountThetaSketch expects at least four arguments (theta-sketch column, parameter(s), post-aggregation expression), got: ",
-        numArguments);
-
-    // Initialize the theta-sketch column
-    _thetaSketchColumn = arguments.get(0);
-    Preconditions.checkArgument(_thetaSketchColumn.getType() == ExpressionContext.Type.IDENTIFIER,
-        "First argument of DistinctCountThetaSketch must be identifier (theta-sketch column)");
-
-    // Initialize the theta-sketch parameters
-    ExpressionContext parametersExpression = arguments.get(1);
-    Preconditions.checkArgument(parametersExpression.getType() == ExpressionContext.Type.LITERAL,
-        "Second argument of DistinctCountThetaSketch must be literal (parameters)");
-    Parameters parameters = new Parameters(parametersExpression.getLiteral());
-
-    // Initialize the theta-sketch set operation builder
-    _setOperationBuilder = new SetOperationBuilder().setNominalEntries(parameters.getNominalEntries());
-
-    // Index of the original input predicates
-    // This list is zero indexed, whereas argument substitution is 1-indexed: index[0] = $1
-    _predicates = new ArrayList<>();
-
-    // Initialize the input expressions
-    // NOTE: It is expected to cover the theta-sketch column and the lhs of the predicates.
-    _inputExpressions = new ArrayList<>();
-    _inputExpressions.add(_thetaSketchColumn);
+    if (numArguments < 4) {
+      // Simple union without post-aggregation
 
-    // Initialize the post-aggregation expression
-    // NOTE: It is modeled as a filter
-    ExpressionContext postAggregationExpression = arguments.get(numArguments - 1);
-    Preconditions.checkArgument(parametersExpression.getType() == ExpressionContext.Type.LITERAL,
-        "Last argument of DistinctCountThetaSketch must be literal (post-aggregation expression)");
-    _postAggregationExpression = QueryContextConverterUtils
-        .getExpression(CalciteSqlParser.compileToExpression(postAggregationExpression.getLiteral()));
+      _inputExpressions = Collections.singletonList(_mainExpression);
+      _includeDefaultSketch = true;
+      _filterEvaluators = Collections.emptyList();
+      _postAggregationExpression = ExpressionContext.forIdentifier(DEFAULT_SKETCH_IDENTIFIER);
+    } else {
+      // Union with post-aggregation
```

Comment:
Just to ensure, we have test for this?

Reply:
Yes, we have unit test and integration test for this